### ISSUE TITLE
intermittent test failure in TickWheelExecutor (in Scala 2.12 community build)

### ISSUE THREAD

**Issue by SethTisue** [http4s/blaze#81]
apparently intermittent, since the problem went away in the next run. I don't recall having seen this before

https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/1683/consoleFull

```
[blaze] 09:03:34.746 [specs2.fixed.env151605868-4] ERROR o.h.blaze.util.TickWheelExecutor - Non-Fatal Exception caught while executing scheduled task
[blaze] java.lang.RuntimeException: Woops!
[blaze] 	at scala.sys.package$.error(package.scala:27) ~[scala-library.jar:1.0.0]
[blaze] 	at org.http4s.blaze.util.TickWheelExecutorSpec$$anon$10.run(TickWheelExecutorSpec.scala:114) ~[test-classes/:na]
[blaze] 	at org.http4s.blaze.util.Execution$$anon$2.execute(Executor.scala:49) ~[classes/:na]
[blaze] 	at org.http4s.blaze.util.TickWheelExecutor.schedule(TickWheelExecutor.scala:117) [classes/:na]
[blaze] 	at org.http4s.blaze.util.TickWheelExecutor.schedule(TickWheelExecutor.scala:82) [classes/:na]
[blaze] 	at org.http4s.blaze.util.TickWheelExecutorSpec.$anonfun$new$33(TickWheelExecutorSpec.scala:116) [test-classes/:na]
[blaze] 	at org.specs2.matcher.MatchResult$$anon$12.$anonfun$asResult$1(MatchResult.scala:343) ~[specs2-matcher-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.ResultExecution.execute(ResultExecution.scala:23) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.ResultExecution.execute$(ResultExecution.scala:21) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.ResultExecution$.execute(ResultExecution.scala:118) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.Result$$anon$10.asResult(Result.scala:229) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.AsResult$.apply(AsResult.scala:25) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.matcher.MatchResult$$anon$12.asResult(MatchResult.scala:343) ~[specs2-matcher-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.AsResult$.apply(AsResult.scala:25) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.specification.core.AsExecution$$anon$1.$anonfun$execute$1(AsExecution.scala:15) ~[specs2-core-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.ResultExecution.execute(ResultExecution.scala:23) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.ResultExecution.execute$(ResultExecution.scala:21) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.ResultExecution$.execute(ResultExecution.scala:118) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.Result$$anon$10.asResult(Result.scala:229) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.AsResult$.apply(AsResult.scala:25) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.AsResult$.$anonfun$safely$1(AsResult.scala:33) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.ResultExecution.execute(ResultExecution.scala:23) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.ResultExecution.execute$(ResultExecution.scala:21) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.ResultExecution$.execute(ResultExecution.scala:118) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.AsResult$.safely(AsResult.scala:33) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.specification.core.Execution$.$anonfun$result$2(Execution.scala:193) ~[specs2-core-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.specification.core.Execution$.$anonfun$withEnv$3(Execution.scala:196) ~[specs2-core-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.ResultExecution.execute(ResultExecution.scala:23) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.ResultExecution.execute$(ResultExecution.scala:21) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.ResultExecution$.execute(ResultExecution.scala:118) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.Result$$anon$10.asResult(Result.scala:229) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.AsResult$.apply(AsResult.scala:25) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.AsResult$.$anonfun$safely$1(AsResult.scala:33) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.ResultExecution.execute(ResultExecution.scala:23) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.ResultExecution.execute$(ResultExecution.scala:21) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.ResultExecution$.execute(ResultExecution.scala:118) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.execute.AsResult$.safely(AsResult.scala:33) ~[specs2-common-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at org.specs2.specification.core.Execution$.$anonfun$withEnv$2(Execution.scala:196) ~[specs2-core-3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418.jar:3.9.3-dbuildxe0a082a24bfaebc9750b10dc18f78c0184b01418]
[blaze] 	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:653) ~[scala-library.jar:na]
[blaze] 	at scala.util.Success.$anonfun$map$1(Try.scala:251) ~[scala-library.jar:1.0.0]
[blaze] 	at scala.util.Success.map(Try.scala:209) ~[scala-library.jar:1.0.0]
[blaze] 	at scala.concurrent.Future.$anonfun$map$1(Future.scala:287) ~[scala-library.jar:na]
[blaze] 	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:29) ~[scala-library.jar:na]
[blaze] 	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:29) ~[scala-library.jar:na]
[blaze] 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60) ~[scala-library.jar:na]
[blaze] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[na:1.8.0_111]
[blaze] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[na:1.8.0_111]
[blaze] 	at java.lang.Thread.run(Thread.java:745) ~[na:1.8.0_111]
[blaze] 09:03:34.750 [TickWheelExecutor: 3 spokes, 1 millisecond interval] ERROR o.h.blaze.util.TickWheelExecutor - Non-Fatal Exception caught while executing scheduled task
[blaze] java.lang.RuntimeException: Woops!
[blaze] 	at scala.sys.package$.error(package.scala:27) ~[scala-library.jar:1.0.0]
[blaze] 	at org.http4s.blaze.util.TickWheelExecutorSpec$$anon$9.run(TickWheelExecutorSpec.scala:108) ~[test-classes/:na]
[blaze] 	at org.http4s.blaze.util.Execution$$anon$2.execute(Executor.scala:49) ~[classes/:na]
[blaze] 	at org.http4s.blaze.util.TickWheelExecutor$Node.run(TickWheelExecutor.scala:272) [classes/:na]
[blaze] 	at org.http4s.blaze.util.TickWheelExecutor$Bucket.checkNext$1(TickWheelExecutor.scala:204) [classes/:na]
[blaze] 	at org.http4s.blaze.util.TickWheelExecutor$Bucket.prune(TickWheelExecutor.scala:212) [classes/:na]
[blaze] 	at org.http4s.blaze.util.TickWheelExecutor.go$3(TickWheelExecutor.scala:161) [classes/:na]
[blaze] 	at org.http4s.blaze.util.TickWheelExecutor.org$http4s$blaze$util$TickWheelExecutor$$cycle(TickWheelExecutor.scala:164) [classes/:na]
[blaze] 	at org.http4s.blaze.util.TickWheelExecutor$$anon$1.run(TickWheelExecutor.scala:56) [classes/:na]
[blaze] 09:03:34.765 [scala-execution-context-global-370] DEBUG org.http4s.blaze.pipeline.Stage - SSL Read Request Status: Status = BUFFER_UNDERFLOW HandshakeStatus = NOT_HANDSHAKING
[blaze] bytesConsumed = 0 bytesProduced = 0, java.nio.HeapByteBuffer[pos=0 lim=16921 cap=16921]
```

**Comment by bryce-anderson** [comment]
Its come up a few times now, so I'll take a look.

**Comment by bryce-anderson** [comment]
I _think_ this was fixed with #95 but please feel free to reopen if not. I suspect the underlying problem is that its just hard to test timeout behavior reliably without sleeping a long time and making the tests super slow.

**Comment by SethTisue** [comment]
thx, I'll keep an eye out
